controller/api/models.py

```diff
@@ -518,7 +541,25 @@ def _default_scale(self, user, release):
 
         self.scale(user, structure)
 
-    def logs(self, log_lines=str(settings.LOG_LINES)):
+    def _get_etcd_client(self):
+        if self._etcd_client is None:
+            try:
+                self._etcd_client = etcd.Client(host=settings.ETCD_HOST,
+                                                port=int(settings.ETCD_PORT))
+            except etcd.EtcdException:
+                logger.log(logging.WARNING,
+                           'Cannot synchronize with etcd cluster')
+        return self._etcd_client
+
+    def _get_logging_mode(self):
+        try:
+            return self._get_etcd_client().get('/deis/logs/handlertype').value
```

Comment:
Also, I'd like to see some of this code simplified and even eliminated.  confd already runs in the background and is used to set values for a number of constants in `settings.py` whenever relevant values in etcd change.  Upon changing, the gunicorn master process for the controller is signaled to reload configuration.  You should take advantage of this existing process to detect configuration changes for "free" without needing to add extra etcd-related code to `models.py`.